src/express/src/services/registryService.ts

```diff
@@ -0,0 +1,39 @@
+import { IRegistry } from '../../../registry/IRegistry.js';
+import type { Logger } from 'pino';
+import { ApiError } from '../errors/ApiError.js';
+
+export class RegistryService {
+  private registry: IRegistry | null = null;
+  private lastRefresh: number = Date.now();
+
+  constructor(
+    private readonly getRegistry: () => Promise<IRegistry>,
+    private readonly refreshInterval: number,
+    private readonly logger: Logger | Console,
+  ) {}
+
+  async initialize() {
+    this.registry = await this.getRegistry();
+  }
+
+  async getCurrentRegistry(): Promise<IRegistry> {
+    const now = Date.now();
+    if (now - this.lastRefresh > this.refreshInterval) {
+      this.logger.info('Refreshing registry cache...');
+      this.registry = await this.getRegistry();
+      this.lastRefresh = now;
+    }
```

Comment:
Maybe it would be a good idea to add an or statement here in case someone forgets to call initialize? `(now - this.lastRefresh > this.refreshInterval) || !this.registry`